docs/release-notes/self-hosted/self-hosted-v12.0.0.md

```diff
@@ -0,0 +1,99 @@
+---
+rss_title: Codacy release notes RSS feed
+rss_href: /feed_rss_created.xml
+description: Release notes for Codacy Self-hosted v12.0.0.
+codacy_tools_version_old: https://github.com/codacy/codacy-tools-sh/releases/tag/sh-1.1.6
+codacy_tools_version_new: https://github.com/codacy/codacy-tools-sh/releases/tag/sh-1.2.1
+---
+
+# Self-hosted v12.0.0
+
+These release notes are for [Codacy Self-hosted v12.0.0](https://github.com/codacy/chart/releases/tag/12.0.0){: target="_blank"}, released on June 21, 2023.<!-- TODO Update release date -->
+
+📢 [Visit the Codacy roadmap](https://roadmap.codacy.com) and <span class="skip-vale">let us know</span> your feedback on both new and planned product updates!
+
+<!--TODO Check these issues manually
+Jira issues without release notes
+-   https://codacy.atlassian.net/browse/TS-110 TODO review
+
+Jira issues with disabled release notes
+-   https://codacy.atlassian.net/browse/TS-369 TODO Notify Paulo that we're ignoring this one: https://codacy.atlassian.net/browse/TS-369?focusedCommentId=56541
+-   https://codacy.atlassian.net/browse/TS-356 TODO we may want to notify this: ESlint: install plugin vue-scoped-css
+-->
+
+## Upgrading Codacy Self-hosted
+
+Follow the steps below to upgrade to Codacy Self-hosted v12.0.0:
+
+1.  Check the [release notes for all Codacy Self-hosted versions](../index.md#self-hosted) **between your current version and the most recent version** for breaking changes and follow the instructions provided <span class="skip-vale">carefully</span>.
+
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v12.0/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the following versions:<!--TODO Update CLI tool versions-->
+
+    -   [Codacy Analysis CLI MAJOR.MINOR.PATCH](https://github.com/codacy/codacy-analysis-cli/releases/tag/MAJOR.MINOR.PATCH)
+    -   [Codacy Coverage Reporter MAJOR.MINOR.PATCH](https://github.com/codacy/codacy-coverage-reporter/releases/tag/MAJOR.MINOR.PATCH)
+
+## Breaking changes
+
```

Comment:
There are some additional breaking changes to be added here, implemented by @lolgab. Will leave a rough draft for us to work on. The following procedure should be performed by everyone upgrading to v12.0.0 if they have set "scheduler.enable = "true" in values.yaml", even if not performing a Kubernetes version update.
The procedure could be done using kubectl but it could be more dangerous and error prone.


We are deprecating custom scheduling functionality.
If you have set scheduler.enable = "true" in values.yaml, before migrating to Codacy Self-hosted v12.0.0 do:
- Remove scheduler settings from values.yaml
- In worker-manager ConfigMap set  `CONFIG_FORCE_codacy_kubernetes_scheduler_name: default-scheduler `
- Restart worker-manager deployment
- Remove any pending worker Pods
"

Reply:
Thanks! Updated as suggested and moving this discussion: https://github.com/codacy/docs/pull/1767/files#r1239917734